bridges/snowbridge/pallets/outbound-queue-v2/src/lib.rs

```diff
@@ -296,17 +307,24 @@ pub mod pallet {
 
 			// Yield if the maximum number of messages has been processed this block.
 			// This ensures that the weight of `on_finalize` has a known maximum bound.
-			ensure!(
-				MessageLeaves::<T>::decode_len().unwrap_or(0) <
-					T::MaxMessagesPerBlock::get() as usize,
-				Yield
-			);
+			let current_len = MessageLeaves::<T>::decode_len().unwrap_or(0);
+			if current_len >= T::MaxMessagesPerBlock::get() as usize {
+				Self::deposit_event(Event::MessageRejected { id: None, nonce: None, error: Yield });
```

Comment:
I don't think we should emit this event here, the message is not rejected in this case, processing is just postponed to next block, no?

For the other errors, the message is actually rejected and forever dropped, and the emitted event can potentially be used as proof of this.

Reply:
Yeah, That's correct. Clara, we don't need to log a rejected event here.